src/source-control/GitSourceControl.ts

```diff
@@ -326,9 +339,10 @@ export class GitSourceControl implements SourceControl {
 
       queue.push(...commitToBeRebased.childCommits);
 
-      // We now would want to use this new commit hash as the target
-      // FIXME: This will turn a commit tree into a linear stack!
-      targetCommitHash = newCommitOid.tostrS();
+      // We now would want to use this new commit hash as the target for all of it's children
+      commitToBeRebased.childCommits.map((childCommitHash: CommitHash) => {
```

Comment:
Nit: prefer `forEach` over `map` when we don't care about the return value. Also since TS already knows the type of `childCommitHash` so it's not strictly necessary to specify it, removing it will reduce the amount of cognitive load imo

```suggestion
      commitToBeRebased.childCommits.forEach((childCommitHash) => {
```